lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -236,8 +236,8 @@ def get_diff_image(self):
                 self._last_renderer, renderer)
             self._force_full = False
             self._png_is_old = False
-
-        return buff
+            self.buff = buff
+        return self.buff
```

Comment:
I guess refresh_all could inspect `self.canvas._png_is_old` to see whether there is a need to refresh, and otherwise just do nothing (no sending of binaries). but that's not entirely clean in terms of attribute access